### PR TITLE
Ria 5496 record the decision

### PR DESCRIPTION
**Pull request by ThomasKKC**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/RIA-5496

### Change description ###
Added record decision notice unsigned document templates

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x ] No
```

### CODE REVIEW

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/bail/helper/BailDecisionUnsignedTemplateHelper.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        fieldValues.put("applicantGivenNames", bailCase.read(APPLICANT_GIVEN_NAMES, String.class));
+        fieldValues.put("applicantFamilyName", bailCase.read(APPLICANT_FAMILY_NAME, String.class));
+
+        if (bailCase.read(IS_ADMIN, YesOrNo.class).orElse(YesOrNo.NO) == YesOrNo.YES) {
+            isLegalRep = (bailCase.read(SENT_BY_CHECKLIST, String.class).orElse("").equalsIgnoreCase("legal representative"));
+        } else if (bailCase.read(IS_LEGAL_REP, YesOrNo.class).orElse(YesOrNo.NO) == YesOrNo.YES) {
+            isLegalRep = true;
+        }
+
+        final boolean isLegallyRepresentedCase = isLegalRep || hasLegalRep;
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/templates/bail/helper/BailDecisionUnsignedTemplateHelper.java:45]: there is ccd field: isLegallyRepresentedForFlag, which can be directly read to see if the case is Legally Represented. Just for future references.

File: src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/bail/BailDecisionUnsignedGrantedCreator.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+        final CaseDetails<BailCase> caseDetails = callback.getCaseDetails();
+        final BailCase bailCase = caseDetails.getCaseData();
+
+        Document bailDocument = bailDocumentCreator.create(caseDetails);
+
+        bailDocumentHandler.addWithMetadata(
+            bailCase,
+            bailDocument,
+            DECISION_UNSIGNED_DOC_METADATA,
```
Comment (neha-aggarw) [inline src/main/java/uk/gov/hmcts/reform/iacasedocumentsapi/domain/handlers/presubmit/bail/BailDecisionUnsignedGrantedCreator.java:70]: TRIBUNAL_DOCUMENTS_WITH_METADATA

**Review by neha-aggarw — CHANGES_REQUESTED** [review]

**Review by neha-aggarw — APPROVED** [review]
LGTM

**Pull request merged**